server/routes/routes.go

```diff
@@ -165,6 +169,8 @@ func v1Routes(ctx context.Context, r gin.IRouter, dexObj *v1.DexObject, localUse
 	r.GET("/namespaces/:namespace/mono-vertices/:mono-vertex/metrics", handler.GetMonoVertexMetrics)
 	// Get the health information of a mono vertex.
 	r.GET("/namespaces/:namespace/mono-vertices/:mono-vertex/health", handler.GetMonoVertexHealth)
+	// Get the time series data for a specific metric.
+	r.POST("/namespaces/:namespace/pipelines/:pipeline/getMetricData", handler.GetMetricData)
```

Comment:
are we planning to cover monovertex with the same endpoint?

Reply:
lets update the comment to include for pipeline for now.

Reply:
Can we provide a query generic query service which works for any metrics?